libcxx/include/__flat_set/utils.h

```diff
@@ -60,7 +61,8 @@ struct __flat_set_utils {
       // C++23 Sequence Container should have insert_range member function
       // Note that not all Sequence Containers provide append_range.
       __set.__keys_.insert_range(__set.__keys_.end(), std::forward<_Range>(__rng));
-    } else if constexpr (ranges::common_range<_Range>) {
+    } else if constexpr (ranges::common_range<_Range> &&
+                         __has_input_iterator_category<ranges::iterator_t<_Range>>::value) {
```

Comment:
No (functional) change requested. Do we want to say the fallback branches are (pedantically) extensions?

Reply:
On the other hand, it seems that the missing parts of `MinSequenceContainer` for C++23 sequence container requirements "silently" enforced our `flat_meow` to contain some extensions.

Reply:
Ah, I guess these cases should be moved to the `libcxx/test/libcxx` subdirectory.